framework/include/utils/MooseMeshUtils.h

```diff
@@ -289,4 +289,28 @@ void makeOrderedNodeList(std::vector<std::pair<dof_id_type, dof_id_type>> & node
                          std::vector<dof_id_type> & elem_id_list,
                          std::vector<dof_id_type> & ordered_node_list,
                          std::vector<dof_id_type> & ordered_elem_id_list);
+
+/**
+ * Converts a given name (BoundaryName or SubdomainName) into a corresponding ID (BoundaryID or
+ * SubdomainID) and performs bounds checking to ensure that overflow doesn't happen.
+ * @param name Name that is to be converted into an ID.
```

Comment:
Add a note that this should only be used on names that are already known to be digits (use Moose::isDigits)